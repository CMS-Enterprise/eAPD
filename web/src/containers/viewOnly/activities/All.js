import PropTypes from 'prop-types';
import React, { Fragment } from 'react';
import { connect } from 'react-redux';

import ActivityList from './ActivityList.js';
import Activity from './Activity.js';

const Activities = ({activities}) => {
  return (
    <div>
      <ActivityList activities={activities} />
      <h2>Activity Details</h2>
      {activities.map((activity) => (
        <Activity activity={activity} />
      ))}
    </div>
  )
}

export default Activities;