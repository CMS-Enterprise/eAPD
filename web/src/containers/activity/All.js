import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import FormAndReviewList from '../../components/FormAndReviewList';
import EntryDetails from './EntryDetails';
import {
  NameAndFundingSourceForm,
  NameAndFundingSourceReview
} from './NameAndFundingSource';
import Waypoint from '../ConnectedWaypoint';
import { addActivity, removeActivity } from '../../actions/editActivity';
import { Section, Subsection } from '../../components/Section';
import { selectAllActivities } from '../../reducers/activities.selectors';

const All = ({ add, first, other, remove }) => {
  const onAdd = () => add();

  const onRemove = index => {
    remove(index);
  };

  return (
    <Waypoint id="activities-overview">
      <Section isNumbered id="activities" resource="activities">
        <Waypoint id="activities-list" />
        <Subsection id="activities-list" resource="activities.list" open>
          <NameAndFundingSourceReview item={first} index={-1} disableExpand />
          <FormAndReviewList
            addButtonText="Add another activity"
            allowDeleteAll
            list={other}
            className="ds-u-border-bottom--0"
            collapsed={NameAndFundingSourceReview}
            expanded={NameAndFundingSourceForm}
            noDataMessage={false}
            onAddClick={onAdd}
            onDeleteClick={onRemove}
          />
        </Subsection>
        {[first, ...other].map(({ key }, index) => (
          <Waypoint id={key} key={key}>
            <EntryDetails activityIndex={index} />
          </Waypoint>
        ))}
      </Section>
    </Waypoint>
  );
};

All.propTypes = {
  add: PropTypes.func.isRequired,
  first: PropTypes.object.isRequired,
  other: PropTypes.arrayOf(PropTypes.object).isRequired,
  remove: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  const activities = selectAllActivities(state);
  return {
    first: activities[0],
    other: activities.slice(1)
  };
};

const mapDispatchToProps = {
  add: addActivity,
  remove: removeActivity
};

export default connect(mapStateToProps, mapDispatchToProps)(All);

export { All as plain, mapStateToProps, mapDispatchToProps };
