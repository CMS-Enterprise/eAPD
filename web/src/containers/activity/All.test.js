import { shallow } from 'enzyme';
import sinon from 'sinon';
import React from 'react';

import { AllRaw as Activities, mapStateToProps } from './All';

describe('the Activities component', () => {
  const props = {
    activityKeys: ['1', '2', '3'],
    addActivity: sinon.stub()
  };

  test('renders correctly', () => {
    const component = shallow(<Activities {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('adds a new activity', () => {
    const component = shallow(<Activities {...props} />);
    component.find('Btn').simulate('click');
    expect(props.addActivity.callCount).toBe(1);
  });

  test('maps redux state to component props', () => {
    expect(
      mapStateToProps({ activities: { allKeys: ['1', 'two', '3'] } })
    ).toEqual({ activityKeys: ['1', 'two', '3'] });
  });
});
