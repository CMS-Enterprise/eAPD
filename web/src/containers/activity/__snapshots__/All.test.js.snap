// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Activities component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <All
    activityKeys={
      Array [
        "1",
        "2",
        "3",
      ]
    }
    addActivity={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Subsection
          id="activities-list"
          open={true}
          resource="activities.list"
        >
          <div
            className="mb3"
          >
            <Connect(EntryBasic)
              aKey="1"
              num={1}
            />
            <Connect(EntryBasic)
              aKey="2"
              num={2}
            />
            <Connect(EntryBasic)
              aKey="3"
              num={3}
            />
          </div>
          <Btn
            extraCss={null}
            kind="primary"
            onClick={[Function]}
            size={null}
          >
            Add activity
          </Btn>
        </Subsection>,
        Array [
          <Connect(EntryDetails)
            aKey="1"
            num={1}
          />,
          <Connect(EntryDetails)
            aKey="2"
            num={2}
          />,
          <Connect(EntryDetails)
            aKey="3"
            num={3}
          />,
        ],
      ],
      "id": "activities",
      "resource": "activities",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="mb3"
            >
              <Connect(EntryBasic)
                aKey="1"
                num={1}
              />
              <Connect(EntryBasic)
                aKey="2"
                num={2}
              />
              <Connect(EntryBasic)
                aKey="3"
                num={3}
              />
            </div>,
            <Btn
              extraCss={null}
              kind="primary"
              onClick={[Function]}
              size={null}
            >
              Add activity
            </Btn>,
          ],
          "id": "activities-list",
          "open": true,
          "resource": "activities.list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(EntryBasic)
                  aKey="1"
                  num={1}
                />,
                <Connect(EntryBasic)
                  aKey="2"
                  num={2}
                />,
                <Connect(EntryBasic)
                  aKey="3"
                  num={3}
                />,
              ],
              "className": "mb3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "aKey": "1",
                  "num": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "aKey": "2",
                  "num": 2,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "class",
                "props": Object {
                  "aKey": "3",
                  "num": 3,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Add activity",
              "extraCss": null,
              "kind": "primary",
              "onClick": [Function],
              "size": null,
            },
            "ref": null,
            "rendered": "Add activity",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "aKey": "1",
          "num": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "aKey": "2",
          "num": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "aKey": "3",
          "num": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Subsection
            id="activities-list"
            open={true}
            resource="activities.list"
          >
            <div
              className="mb3"
            >
              <Connect(EntryBasic)
                aKey="1"
                num={1}
              />
              <Connect(EntryBasic)
                aKey="2"
                num={2}
              />
              <Connect(EntryBasic)
                aKey="3"
                num={3}
              />
            </div>
            <Btn
              extraCss={null}
              kind="primary"
              onClick={[Function]}
              size={null}
            >
              Add activity
            </Btn>
          </Subsection>,
          Array [
            <Connect(EntryDetails)
              aKey="1"
              num={1}
            />,
            <Connect(EntryDetails)
              aKey="2"
              num={2}
            />,
            <Connect(EntryDetails)
              aKey="3"
              num={3}
            />,
          ],
        ],
        "id": "activities",
        "resource": "activities",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="mb3"
              >
                <Connect(EntryBasic)
                  aKey="1"
                  num={1}
                />
                <Connect(EntryBasic)
                  aKey="2"
                  num={2}
                />
                <Connect(EntryBasic)
                  aKey="3"
                  num={3}
                />
              </div>,
              <Btn
                extraCss={null}
                kind="primary"
                onClick={[Function]}
                size={null}
              >
                Add activity
              </Btn>,
            ],
            "id": "activities-list",
            "open": true,
            "resource": "activities.list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(EntryBasic)
                    aKey="1"
                    num={1}
                  />,
                  <Connect(EntryBasic)
                    aKey="2"
                    num={2}
                  />,
                  <Connect(EntryBasic)
                    aKey="3"
                    num={3}
                  />,
                ],
                "className": "mb3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "aKey": "1",
                    "num": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "aKey": "2",
                    "num": 2,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "class",
                  "props": Object {
                    "aKey": "3",
                    "num": 3,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Add activity",
                "extraCss": null,
                "kind": "primary",
                "onClick": [Function],
                "size": null,
              },
              "ref": null,
              "rendered": "Add activity",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "aKey": "1",
            "num": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "aKey": "2",
            "num": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "aKey": "3",
            "num": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
